src/bundles/SellerRegistration/components/DocumentsForm/DocumentsForm.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Form, actions } from 'react-redux-form';
+
+import 'whatwg-fetch';
+
+//import { required } from '../../../../validators';
+
+import Layout        from '../../../../shared/Layout';
+import BaseForm      from '../../../../shared/form/BaseForm';
+import SubmitForm    from '../../../../shared/form/SubmitForm';
+import ErrorBox      from '../../../../shared/form/ErrorBox';
+import StatefulError from '../../../../shared/form/StatefulError';
+
+import formProps     from '../../../../shared/reduxModules/formPropsSelector';
+
+class DocumentsForm extends BaseForm {
+
+  static propTypes = {
+    action: React.PropTypes.string,
+    csrf_token: React.PropTypes.string,
+    form: React.PropTypes.object.isRequired,
+    dispatch: React.PropTypes.func.isRequired,
+    model: React.PropTypes.string.isRequired
+  }
+
+  state = {
+    errors: {}
+  }
+
+  files = [
+    { 'label': 'Finanical Statement', 'id': 'financial' },
+    { 'label': 'Public Liability Insurance', 'id': 'liability' },
+    { 'label': 'Workers Compensation Insurance', 'id': 'workers' },
+    { 'label': 'Signed copy of Deed of Standing Offer', 'id': 'deed' },
+  ]
+
+  onUpload(id, e) {
+    e.preventDefault();
+    const { dispatch, model, action, csrf_token } = this.props;
+    const file = this.state[id].file;
+
+    this.setState({
+      [id]: Object.assign({}, this.state[id], {'uploading': true, 'file': void 0}),
+      errors: Object.assign({}, this.state.errors, {[id]: void 0})
+    })
+
+    dispatch(actions.omit(`${model}.documents`, id))
+
+    let data = new FormData()
+    data.append(id, file)
+    fetch(`${action}/${id}`, {
```

Comment:
i thought that you guys had api available in dispatch to handle fetch src/shared/reduxModules/api.js. Also, this could be an action?